client/src/types/index.ts

```diff
@@ -10,6 +11,59 @@ export type Asset = {
   organization: number;
 };
 
+export type Organization = {
+  id?: number;
+  name: string;
+  doing_business_as: string;
+  description: string;
+  website: string;
+  address: string;
+  phone: string;
+  city: string;
+  state: string;
+  ein: string;
+  nonprofit_classification: string;
+};
+
+export type UserOrg = {
+  id?: number;
+  approvalStatus: ApprovalStatus;
+  role: Role;
+};
+
+export type UserOrgCreateObj = {
+  organization: { id: number };
+  user: { id: number };
+  approvalStatus: ApprovalStatus;
+  role: Role;
+};
+
+export enum Role {
+  admin = 'ADMIN',
+  owner = 'OWNER',
+  revoked = 'REVOKED',
+}
+
+export enum ApprovalStatus {
+  approved = 'APPROVED',
+  pending = 'PENDING',
+  denied = 'DENIED',
+}
+
+export type UserEntity = {
+  id?: number;
+  firstName: string;
+  last_name: string;
+  email: string;
+};
+
+export type UserEntityCreateObj = {
+  firstName: string;
+  last_name: string;
+  email: string;
+  password: string;
+};
+
```

Comment:
all this redundancy is not good. let's have:
```
type BaseUser = {
  firstName: string;
  last_name: string;
  email: string;
};

type User = BaseUser & { id: string };

type NewUser = BaseUser & { password: string };
```
or something along these lines

Reply:
Thanks! that is a great point. I moved one to the one place it is used and made it local. I refactored to other 2 to avoid repetition.